bug: MOM6 model mod == vs. .eqv.

:bug: 🪰  Your bug may already be reported!
Please search on the [issue tracker](https://github.com/NCAR/DART/issues) before creating a new issue.

### Describe the bug

1. Compile with GNU Fortran (GCC) 12.2.0 20220819  or Cray ftn
2. What was the expected outcome? sucess
3. What actually happened?  fail
    ```
    /glade/derecho/scratch/hkershaw/build_everything/gcc.2023-12-08T1147/DART/models/MOM6/model_mod.f90:330:8:

   330 | if (any(found == .false.)) then
       |        1
   Error: Logicals at (1) must be compared with .eqv. instead of ==
   make: *** [Makefile:62: model_mod.o] Error 1
   ```

### Which model(s) are you working with?
MOM6

### Version of DART      
v10.9.2

### Have you modified the DART code?   
No  

### Build information
Please describe:  
 1. Derecho 
 2. gnu GNU Fortran (GCC) 12.2.0 20220819 (HPE)

--- a/models/MOM6/model_mod.f90
+++ b/models/MOM6/model_mod.f90
@@ -327,7 +327,7 @@ FIND_LAYER: do i = 2, nz
 
 enddo FIND_LAYER
 
-if (any(found == .false.)) then
+if (any(found .eqv. .false.)) then
    istatus(:) = OBS_TOO_DEEP
    return
 endif

